lib/import-map.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Add import map support.
+ *
+ * @package gutenberg
+ */
+
+$import_map_entries = array();
+
+/**
+ * Registers an import map entry.
+ *
+ * @since 5.6.0
+ *
+ * @global array $import_map_entries The registered import map entries.
+ *
+ * @param string $key   Import map item key.
+ * @param string $value Import map item value.
+ */
+function gutenberg_register_import_map_entry( $key, $value ) {
+	global $import_map_entries;
+
+	$import_map_entries[ $key ] = $value;
+}
+
+/**
+ * Injects the import map to admin pages.
+ */
+function gutenberg_inject_import_map() {
+	global $import_map_entries;
+
+	if ( empty( $import_map_entries ) ) {
+		return;
+	}
+
+	echo '
+<script type="importmap">
+{
+	"imports": ' . json_encode( $import_map_entries ) . '
+}
+</script>
+	';
+}
+
+add_action( 'admin_enqueue_scripts', 'gutenberg_inject_import_map' );
```

Comment:
The import map is injected at the top of all WP pages (admin for now, but could be added to frontend too)